docs/fxa-engineering/fxa-oauth-webchannel-protocol.md

```diff
@@ -0,0 +1,52 @@
+---
+id: fxa-oauth-webchannel-protocol
+title: FxA/OAuth WebChannel Protocol
+sidebar_label: OAuth WebChannel Protocol
+---
+
+Current as of `December 2nd, 2019`
+
+# Communication with OAuth WebChannels
+
+OAuth WebChannels is an extension of the [FxA WebChannel Desktop Flow](./fxa-webchannel-protocol).
+It consists of similar messages as the Desktop flow. Some behaviours in the OAuth flow are different.
+
+This flow currently supports the following messages:
+
+-   `fxaccounts:fxa_status`
+-   `fxaccounts:oauth_login`
+
+The `fxa_status` message in the OAuth flow supports specifying a custom list of Sync engines that the app supports.
+See [FxA WebChannel Desktop Flow](./fxa-webchannel-protocol) for details of engine capabilities.
+
+## Communication with GeckoView applications
+
+To enable this feature in applications with GeckoView we ship a WebExtension
+as part of the [firefox-accounts](https://github.com/mozilla-mobile/android-components/blob/master/components/service/firefox-accounts/README.md) Android component.
```

Comment:
@vladikoff I wonder if there are any other limitations/gotchas with this webextension that should be called out here. I'm thinking of the thing where we can't easily point it at stage or dev because the webextension has to be configured to inject in a specific origin, but I don't remember the details.